roachtest: restore in `drop` test failing to run

The test calls an `ALTER` on the `tpcc.order_line` table, which fails because the workload command to restore that table from a fixture seems never to have been run. @benesch and I made the roachtest process `os.Exit` before wiping the cluster so we could inspect the admin UI, and saw that no jobs had been run. (IIRC)

Happens in CI (e.g. https://teamcity.cockroachdb.com/viewLog.html?tab=buildLog&buildTypeId=Cockroach_Nightlies_WorkloadNightly&buildId=587077&_focus=933), but haven't been able to repro on a roachprod cluster launched from a local machine.

### roachtest output:

```
[20:37:56]--- FAIL: drop/tpcc/w=100,nodes=9 (378.27s)
[20:37:56]	drop.go:52,drop.go:64,cluster.go:784,errgroup.go:58: pq: relation "tpcc.order_line" does not exist
```

### `test.log`:
```
> roachprod create teamcity-587077-drop-tpcc-w-100-nodes-9 -n 9 --gce-machine-type=n1-standard-4 --gce-zones=us-central1-b,us-west1-b,europe-west2-b
Creating cluster teamcity-587077-drop-tpcc-w-100-nodes-9 with 9 nodes
OK
teamcity-587077-drop-tpcc-w-100-nodes-9: [gce] 12h28m1s remaining
  teamcity-587077-drop-tpcc-w-100-nodes-9-0001	teamcity-587077-drop-tpcc-w-100-nodes-9-0001.us-central1-b.cockroach-ephemeral	10.128.0.38	35.226.236.65
  teamcity-587077-drop-tpcc-w-100-nodes-9-0002	teamcity-587077-drop-tpcc-w-100-nodes-9-0002.us-central1-b.cockroach-ephemeral	10.128.0.36	35.224.126.22
  teamcity-587077-drop-tpcc-w-100-nodes-9-0003	teamcity-587077-drop-tpcc-w-100-nodes-9-0003.us-central1-b.cockroach-ephemeral	10.128.0.35	35.224.122.201
  teamcity-587077-drop-tpcc-w-100-nodes-9-0004	teamcity-587077-drop-tpcc-w-100-nodes-9-0004.us-central1-b.cockroach-ephemeral	10.128.0.41	35.225.55.255
  teamcity-587077-drop-tpcc-w-100-nodes-9-0005	teamcity-587077-drop-tpcc-w-100-nodes-9-0005.us-central1-b.cockroach-ephemeral	10.128.0.42	35.225.29.213
  teamcity-587077-drop-tpcc-w-100-nodes-9-0006	teamcity-587077-drop-tpcc-w-100-nodes-9-0006.us-central1-b.cockroach-ephemeral	10.128.0.27	35.224.74.240
  teamcity-587077-drop-tpcc-w-100-nodes-9-0007	teamcity-587077-drop-tpcc-w-100-nodes-9-0007.us-central1-b.cockroach-ephemeral	10.128.0.40	35.225.51.246
  teamcity-587077-drop-tpcc-w-100-nodes-9-0008	teamcity-587077-drop-tpcc-w-100-nodes-9-0008.us-central1-b.cockroach-ephemeral	10.128.0.39	35.224.233.183
  teamcity-587077-drop-tpcc-w-100-nodes-9-0009	teamcity-587077-drop-tpcc-w-100-nodes-9-0009.us-central1-b.cockroach-ephemeral	10.128.0.37	35.224.227.144
2018/03/29 20:31:58 could not clear ssh key for hostname 35.226.236.65:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.224.126.22:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.224.122.201:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.225.55.255:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.225.29.213:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.224.74.240:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.225.51.246:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.224.233.183:
do_known_hosts: hostkeys_foreach failed: No such file or directory
2018/03/29 20:31:58 could not clear ssh key for hostname 35.224.227.144:
do_known_hosts: hostkeys_foreach failed: No such file or directory
Syncing...
teamcity-587077-drop-tpcc-w-100-nodes-9: waiting for nodes to start.............................................................
> roachprod put teamcity-587077-drop-tpcc-w-100-nodes-9:1-9 /home/agent/work/.go/src/github.com/cockroachdb/cockroach/cockroach-linux-2.6.32-gnu-amd64 ./cockroach
teamcity-587077-drop-tpcc-w-100-nodes-9: putting /home/agent/work/.go/src/github.com/cockroachdb/cockroach/cockroach-linux-2.6.32-gnu-amd64 ./cockroach
.................
   1: done
   2: done
   3: done
   4: done
   5: done
   6: done
   7: done
   8: done
   9: done
> roachprod put teamcity-587077-drop-tpcc-w-100-nodes-9:1-9 /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/workload ./workload
teamcity-587077-drop-tpcc-w-100-nodes-9: putting /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/workload ./workload
................
   1: done
   2: done
   3: done
   4: done
   5: done
   6: done
   7: done
   8: done
   9: done
> roachprod start -e COCKROACH_MEMPROF_INTERVAL=15s teamcity-587077-drop-tpcc-w-100-nodes-9:1-9
teamcity-587077-drop-tpcc-w-100-nodes-9: starting........
teamcity-587077-drop-tpcc-w-100-nodes-9: initializing cluster settings
SET CLUSTER SETTING
> roachprod run teamcity-587077-drop-tpcc-w-100-nodes-9:1 -- ./workload fixtures load tpcc --warehouses=100 --db tpcc {pgurl:1}
MONITOR: 4: 2683
MONITOR: 8: 2692
MONITOR: 1: 2707
MONITOR: 7: 2754
MONITOR: 5: 2723
MONITOR: 9: 2746
MONITOR: 6: 2725
MONITOR: 3: 2701
MONITOR: 2: 2721
> roachprod get teamcity-587077-drop-tpcc-w-100-nodes-9 logs /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20180329-587077/drop/tpcc/w=100,nodes=9/logs
teamcity-587077-drop-tpcc-w-100-nodes-9: getting logs /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20180329-587077/drop/tpcc/w=100,nodes=9/logs
..
   1: done
   2: done
   3: done
   4: done
   5: done
   6: done
   7: done
   8: done
   9: done
> roachprod destroy teamcity-587077-drop-tpcc-w-100-nodes-9
Destroying cluster teamcity-587077-drop-tpcc-w-100-nodes-9 with 9 nodes
OK
```

--- a/pkg/cmd/roachtest/drop.go
+++ b/pkg/cmd/roachtest/drop.go
@@ -40,11 +40,19 @@ func init() {
 		c.Put(ctx, workload, "./workload", c.Range(1, nodes))
 		c.Start(ctx, c.Range(1, nodes), startArgs("-e", "COCKROACH_MEMPROF_INTERVAL=15s"))
 
-		db := c.Conn(ctx, 1)
-		defer db.Close()
-
 		m := newMonitor(ctx, c, c.Range(1, nodes))
 		m.Go(func(ctx context.Context) error {
+			t.Status("importing TPCC fixture")
+			c.Run(ctx, c.Node(1), fmt.Sprintf(
+				"./workload fixtures load tpcc --warehouses=%d --db tpcc {pgurl:1}", warehouses))
+
+			// Don't open the DB connection until after the data has been imported.
+			// Otherwise the ALTER TABLE query below might fail to find the
+			// tpcc.order_line table that we just imported (!) due to what seems to
+			// be a problem with table descriptor leases (#24374).
+			db := c.Conn(ctx, 1)
+			defer db.Close()
+
 			run := func(stmt string) {
 				t.Status(stmt)
 				_, err := db.ExecContext(ctx, stmt)
@@ -54,10 +62,6 @@ func init() {
 			}
 
 			run(`SET CLUSTER SETTING trace.debug.enable = true`)
-
-			t.Status("importing TPCC fixture")
-			c.Run(ctx, c.Node(1), fmt.Sprintf(
-				"./workload fixtures load tpcc --warehouses=%d --db tpcc {pgurl:1}", warehouses))
 
 			// Drop a constraint that would get in the way of deleting from tpcc.stock.
 			const stmtDropConstraint = "ALTER TABLE tpcc.order_line DROP CONSTRAINT fk_ol_supply_w_id_ref_stock"

